src/AeroSharp/DataAccess/MapAccess/Parsers/IMapBinParser.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Aerospike.Client;
+using System.Collections.Generic;
+
+namespace AeroSharp.DataAccess.MapAccess.Parsers;
+
+/// <summary>
+///     Parses generic map data from Aerospike records as <see cref="KeyValuePair{TKey,TValue}"/>s.
+/// </summary>
+internal interface IMapBinParser
+{
+    /// <summary>
+    ///     Parses a single map entry from the given record as a <see cref="KeyValuePair{TKey,TValue}"/>.
+    /// </summary>
+    /// <param name="record"> The record to parse. </param>
+    /// <param name="bin"> The bin to parse. </param>
+    /// <returns> The parsed map entry. </returns>
+    KeyValuePair<object, object> ParseOne(Record record, string bin);
```

Comment:
Wasn't sure what the best way to share this was, so here's quick screenshot.

![image](https://github.com/wayfair-incubator/AeroSharp/assets/45316999/79dc019f-5f97-4cc2-bb99-9b9894242c97)

What's stored on the returned map record seems to always be `KeyValuePair<object, object>` regardless of what types you've written to the map. I don't love it and didn't have an idea of what would be better here. What do you think?

Reply:
Huh, maybe something that would need addressed lower in the core library instead of at the Aerosharp level. Though, I'm not confident that any solution would be backwards compatible.

Sad times, but thanks for checking-- I figured you did it for good reason.